scipy/interpolate/fitpack/pardtc.f

```diff
@@ -0,0 +1,155 @@
+c    Cross Campus Road, Rondebosch 7700, Cape Town, South Africa.
+c    e-mail : cong.ma@uct.ac.za
+c
+c  latest update : may 2019
+c
+c  ..scalar arguments..
+      integer nx,ny,kx,ky,nux,nuy,ier
+c  ..array arguments..
+      real*8 tx(nx),ty(ny),c((nx-kx-1)*(ny-ky-1)),
+     * newc((nx-kx-1)*(ny-ky-1))
+c  ..local scalars..
+      integer i,j,kx1,ky1,lx,ly,l1,l2,m,m0,m1,
+     * nnewc,nkx1,nky1,nxx,nyy,newkx,newky
+      real*8 ak,fac
+c  ..
+c  before starting computations a data check is made. if the input data
+c  are invalid control is immediately repassed to the calling program.
+      ier = 10
+      if(nux.lt.0 .or. nux.ge.kx) go to 400
+      if(nuy.lt.0 .or. nuy.ge.ky) go to 400
+      kx1 = kx+1
+      ky1 = ky+1
+      nkx1 = nx-kx1
+      nky1 = ny-ky1
+      nc = nkx1*nky1
+      ier = 0
+      nxx = nkx1
+      nyy = nky1
+      newkx = kx
+      newky = ky
+c  the partial derivative of order (nux,nuy) of a bivariate spline of
+c  degrees kx,ky is a bivariate spline of degrees kx-nux,ky-nuy.
+c  we calculate the b-spline coefficients of this spline
+c  that is to say newkx = kx - nux, newky = ky - nuy
+      do 70 i=1,nc
+        newc(i) = c(i)
+  70  continue
+      if(nux.eq.0) go to 200
+      lx = 1
+      do 100 j=1,nux
+        ak = newkx
+        nxx = nxx-1
+        l1 = lx
+        m0 = 1
+        do 90 i=1,nxx
+          l1 = l1+1
+          l2 = l1+newkx
+          fac = tx(l2)-tx(l1)
+          if(fac.le.0.) go to 90
+          do 80 m=1,nyy
+            m1 = m0+nyy
+            newc(m0) = (newc(m1)-newc(m0))*ak/fac
+            m0  = m0+1
+  80      continue
+  90    continue
+        lx = lx+1
+        newkx = newkx-1
+ 100  continue
+ 200  if(nuy.eq.0) go to 400
+c orig: if(nuy.eq.0) go to 300
+      ly = 1
+      do 230 j=1,nuy
+        ak = newky
+        nyy = nyy-1
+        l1 = ly
+        do 220 i=1,nyy
+          l1 = l1+1
+          l2 = l1+newky
+          fac = ty(l2)-ty(l1)
+          if(fac.le.0.) go to 220
+          m0 = i
+          do 210 m=1,nxx
+            m1 = m0+1
+            newc(m0) = (newc(m1)-newc(m0))*ak/fac
+            m0  = m0+nky1
+ 210      continue
+ 220    continue
+        ly = ly+1
+        newky = newky-1
+ 230  continue
+      m0 = nyy
+      m1 = nky1
+      do 250 m=2,nxx
+        do 240 i=1,nyy
+          m0 = m0+1
+          m1 = m1+1
+          newc(m0) = newc(m1)
+ 240    continue
+        m1 = m1+nuy
+ 250  continue
+c300  iwx = 1+nxx*nyy
+c     iwy = iwx+mx*(kx1-nux)
+c
+c from parder.f:
+c     call fpbisp(tx(nux+1),nx-2*nux,ty(nuy+1),ny-2*nuy,newc,newkx,newky,
+c    * x,mx,y,my,z,newc(iwx),newc(iwy),iwrk(1),iwrk(mx+1))
+c
+c from bispev.f:
+c     call fpbisp(tx,       nx,      ty,       ny,      c,   kx,   ky,
+c    * x,mx,y,my,z,wrk(1),   wrk(iw),  iwrk(1),iwrk(mx+1))
+c
+c from fpbisp.f:
+c          fpbisp(tx,       nx,      ty,       ny,      c,   kx,   ky,
+c    * x,mx,y,my,z,wx,       wy,       lx,     ly)
```

Comment:
These comments may go or do they serve a purpose? Please add a comment if it's the latter.